docs/src/main/asciidoc/reactive-postgres-client.adoc

```diff
@@ -0,0 +1,450 @@
+@Consumes("application/json")
+public class FruitResource {
+
+    @Inject
+    io.reactiverse.axle.pgclient.PgPool client;
+}
+----
+
+== Database schema and seed data
+
+Before we implement the REST endpoint and data management code, we need to setup the database schema.
+It would also be convenient to have some data inserted upfront.
+
+For production we would recommend to use something like the link:flyway-guide.html[Flyway database migration tool].
+But for development we can simply drop and create the tables on startup, and then insert a few fruits.
+
+[source,java]
+.src/main/java/org/acme/vertx/FruitResource.java
+----
+    @Inject
+    @ConfigProperty(name = "myapp.schema.create", defaultValue = "true") // <1>
+    boolean schemaCreate;
+
+    @PostConstruct
+    void config() {
+        if (schemaCreate) {
+            initdb();
+        }
+    }
+
+    private void initdb() {
+        // TODO
+    }
+}
+----
+
+TIP: You may override the default value of the `myapp.schema.create` property in the `application.properties` file.
+
+Almost ready!
+To initialize the DB in development mode, we will use the client simple `query` method.
+It returns a `CompletionStage` and thus can be composed to execute queries sequentially:
+
+[source,java]
+----
+        client.query("DROP TABLE IF EXISTS fruits")
+                .thenCompose(r -> client.query("CREATE TABLE fruits (id SERIAL PRIMARY KEY, name TEXT NOT NULL)"))
+                .thenCompose(r -> client.query("INSERT INTO fruits (name) VALUES ('Orange')"))
+                .thenCompose(r -> client.query("INSERT INTO fruits (name) VALUES ('Pear')"))
+                .thenCompose(r -> client.query("INSERT INTO fruits (name) VALUES ('Apple')"))
+                .toCompletableFuture()
+                .join();
+----
+
+NOTE: Wondering why we need to convert the result to `CompletableFuture` and block until the latest query is completed?
+This code is part of a `@PostConstruct` method and {project-name} invokes it synchronously.
+As a consequence, returning prematurely could lead to serving requests while the database is not ready yet.
+
+That's it!
+So far we have seen how to configure a pooled client and execute simple queries.
+We are now ready to develop the data management code and implement our RESTful endpoint.
+
+== Using
+
+=== Query results traversal
+
+In development mode, the database is set up with a few rows in the `fruits` table.
+To retrieve all the data, we will use the `query` method again:
+
+[source,java]
+----
+CompletionStage<PgRowSet> rowSet = client.query("SELECT id, name FROM fruits ORDER BY name ASC");
+----
+
+When the operation completes, we will get a `PgRowSet` that has all the rows buffered in memory.
+It can be traversed with an `Iterator`:
+
+[source,java]
+----
+CompletionStage<List<Fruit>> fruits = rowSet.thenApply(pgRowSet -> {
+    List<Fruit> list = new ArrayList<>(pgRowSet.size());
+    PgIterator pgIterator = pgRowSet.iterator();
```

Comment:
We _really_ need to fix codegen to generate `Iterable` objects. This really sucks, no matter how many times I've been bitten by this.

Reply:
Is there a vertx issue open for this @tsegismont ?

Reply:
https://github.com/vert-x3/vertx-codegen/issues/252

I've planned to fix it in 3.7.1